tests/func/test_add.py

```diff
@@ -662,3 +662,39 @@ def test_not_raises_on_re_add(tmp_dir, dvc):
 
     tmp_dir.gen({"file2": "file2 content", "file": "modified file"})
     dvc.add(["file2", "file"])
+
+
+@pytest.mark.parametrize("link", ["hardlink", "symlink", "copy"])
+def test_add_empty_files(tmp_dir, dvc, link):
+    file = "foo"
+    dvc.cache.local.cache_types = [link]
+    stages = tmp_dir.dvc_gen(file, "")
+
+    assert (tmp_dir / file).exists()
+    assert (tmp_dir / (file + Stage.STAGE_FILE_SUFFIX)).exists()
+    assert os.path.exists(stages[0].outs[0].cache_path)
+
+
+@mock.patch(
+    "dvc.remote.local.RemoteLOCAL.is_hardlink",
+    side_effect=RemoteLOCAL.is_hardlink,
+)
```

Comment:
You can use `mocker.spy` for that. Sample usage:
`test_should_remove_outs_before_import`.

Reply:
Thanks @pared. Adjusted. :fire: